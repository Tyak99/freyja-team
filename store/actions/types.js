export const INIT_AUTH_REQUEST = 'INIT_AUTH_REQUEST';

export const END_AUTH_REQUEST = 'END_AUTH_REQUEST';

export const SET_CURRENT_USER = 'SET_CURRENT_USER';

export const SIGNUP_ERROR = 'SIGNUP_ERROR';

export const LOGIN_ERROR = 'LOGIN_ERROR';
