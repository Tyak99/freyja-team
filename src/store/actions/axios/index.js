import axios from 'axios';


const instance = axios.create({
  baseURL: 'https://freyja-ah-backend.herokuapp.com/',
});

instance.interceptors.request.use(
  (config) => {
    // get token expiry and compare with time now

    const token = localStorage.getItem('token');
    config.headers.authorization = token;
    return config;
  },
  error => Promise.reject(error),
);

instance.interceptors.response.use(
  response => response,

  error => {
    const { data } = error.response;
    if (data.status === 401 && (data.error === "No Authentication Token Provided" || data.error === "token not verified")) {
      window.href = '/login';
    }
    return Promise.reject(error),
  }
);

export default instance;