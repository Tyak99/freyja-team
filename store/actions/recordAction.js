var meter = null;
var WIDTH = 700;
var recordingStarted = false;
​
// // initialize SpeechRecognition object
let recognition = new webkitSpeechRecognition();
recognition.maxAlternatives = 1;
recognition.continuous = true;
​
// // Detect the said words
recognition.onresult = e => {
  var current = event.resultIndex;
​
//   // Get a transcript of what was said.
  var transcript = event.results[current][0].transcript;
​
//   // Add the current transcript with existing said values
  var noteContent = $('#saidwords').val();
  noteContent += ' ' + transcript;
  $('#saidwords').val(noteContent);
};
​
// Stop recording
function stopSpeech() {
  // Change status
  $('#status').text('Recording Stopped.');
  recordingStarted = false;
​
  // Stop recognition
  recognition.stop();
}
​
// Start recording
function startSpeech() {
  try {
    // calling it twice will throw..
    $('#status').text('Recording Started.');
    $('#saidwords').val('');
    recordingStarted = true;
​
    // Start recognition
    recognition.start();
  } catch (e) {}
}
​
navigator.getUserMedia({ audio: true }, startUserMedia, function(e) {
  __log('No live audio input: ' + e);
});
​
function startUserMedia(stream) {
  const ctx = new AudioContext();
  const analyser = ctx.createAnalyser();
  const streamNode = ctx.createMediaStreamSource(stream);
  streamNode.connect(analyser);
​
  // Create a new volume meter and connect it.
  meter = createAudioMeter(ctx);
  streamNode.connect(meter);
​
  drawLoop();
}
​
// Create pitch bar
function drawLoop(time) {
  var pitchVolume = meter.volume * WIDTH * 1.4;
​
  var width = 0;
​
  // Pitch detection minimum volume
  var minimum_volume = 150;
​
  // Get width if Recording started
  if (recordingStarted) {
    if (pitchVolume < minimum_volume) {
      width = 0;
    } else if (
      pitchVolume >= minimum_volume &&
      pitchVolume < minimum_volume + 20
    ) {
      width = 10;
    } else if (
      pitchVolume >= minimum_volume + 20 &&
      pitchVolume < minimum_volume + 40
    ) {
      width = 20;
    } else if (
      pitchVolume >= minimum_volume + 40 &&
      pitchVolume < minimum_volume + 60
    ) {
      width = 30;
    } else if (
      pitchVolume >= minimum_volume + 60 &&
      pitchVolume < minimum_volume + 80
    ) {
      width = 40;
    } else if (
      pitchVolume >= minimum_volume + 80 &&
      pitchVolume < minimum_volume + 100
    ) {
      width = 50;
    } else if (
      pitchVolume >= minimum_volume + 100 &&
      pitchVolume < minimum_volume + 120
    ) {
      width = 60;
    } else if (
      pitchVolume >= minimum_volume + 120 &&
      pitchVolume < minimum_volume + 140
    ) {
      width = 70;
    } else if (
      pitchVolume >= minimum_volume + 140 &&
      pitchVolume < minimum_volume + 160
    ) {
      width = 80;
    } else if (
      pitchVolume >= minimum_volume + 160 &&
      pitchVolume < minimum_volume + 200
    ) {
      width = 90;
    } else if (pitchVolume >= minimum_volume + 500) {
      width = 100;
    }
  }
​
  // Update width
  document.getElementById('voiceVolume').style.width = width + '%';
​
  rafID = window.requestAnimationFrame(drawLoop);
}