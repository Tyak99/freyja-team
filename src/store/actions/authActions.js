import axios from 'axios';
import {
  SET_CURRENT_USER,
  LOGIN_ERROR,
  INIT_AUTH_REQUEST,
  END_AUTH_REQUEST,
} from './types';

const authUrl = 'https://freyja-ah-backend.herokuapp.com/api/';

export const setCurrentUser = payload => ({
  type: SET_CURRENT_USER,
  payload,
});

export const loginUser = payload => (dispatch) => {
  dispatch({ type: INIT_AUTH_REQUEST });

  axios.post(`${authUrl}users/login`, payload)
    .then((res) => {
      dispatch({ type: END_AUTH_REQUEST });
      const { token, user } = res.data;

      localStorage.setItem('token', token);
      dispatch(setCurrentUser(user));
    })
    .catch((err) => {
      dispatch({ type: END_AUTH_REQUEST });
      const { data } = err.response;

      const errorObj = Array.isArray(data.error) ? { error: data.error[0].msg } : data;
      dispatch({
        type: LOGIN_ERROR,
        payload: errorObj,
      });
    });
};
