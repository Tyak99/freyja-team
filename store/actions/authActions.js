import axios from 'axios';
import jwtDecode from 'jwt-decode';
import {
  SET_CURRENT_USER,
  LOGIN_ERROR,
  INIT_AUTH_REQUEST,
  END_AUTH_REQUEST,
} from './types';
import setAuthToken from '../../src/utils/setAuthToken';

const authUrl = 'https://freyja-ah-backend.herokuapp.com/api/';

export const setCurrentUser = payload => ({
  type: SET_CURRENT_USER,
  payload,
});

export const loginUser = payload => async (dispatch) => {
  dispatch({ type: INIT_AUTH_REQUEST });

  try {
    const request = await axios.post(`${authUrl}users/login`, payload);
    dispatch({
      type: END_AUTH_REQUEST,
    });

    const { token } = request.data;
    localStorage.setItem('token', token);

    setAuthToken(token);
    const decoded = jwtDecode(token);
    dispatch(setCurrentUser(decoded));
  } catch (error) {
    dispatch({
      type: END_AUTH_REQUEST,
    });

    const { data } = error.response;
    const errorObj = Array.isArray(data.error) ? { error: data.error[0].msg } : data;
    dispatch({
      type: LOGIN_ERROR,
      payload: errorObj,
    });
  }
};
