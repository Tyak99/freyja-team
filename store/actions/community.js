import { get } from 'axios';
import { GET_COMMUNITY_MESSAGES, GET_COMMUNITY_MESSAGES_ERROR, LOADING } from './types';

export const communityMessageSuccess = data => ({
  type: GET_COMMUNITY_MESSAGES,
  data,
});

export const communityMessageError = data => ({
  type: GET_COMMUNITY_MESSAGES_ERROR,
  data,
});

export const loading = () => ({
  type: LOADING,
});

export const getCommunityMessages = () => async (dispatch) => {
  dispatch(loading());
  try {
    const response = await get('https://freyja-ah-backend.herokuapp.com/api/community/messages', {
      headers: { Authorization: `${localStorage.getItem('token')}` },
    });
    dispatch(communityMessageSuccess(response.data.data));
  } catch (error) {
    dispatch(communityMessageError('internal error'));
  }
};
