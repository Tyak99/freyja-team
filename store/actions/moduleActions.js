import { get } from 'axios';
import {
  GET_ALL_MODULES, GET_SINGLE_MODULE, MODULES_ERROR, INIT_MODULE_REQUEST,
} from './types';

const baseUrl = 'https://freyja-ah-backend.herokuapp.com/api/';

const getModules = () => async (dispatch) => {
  try {
    const modulesRequest = await get('https://freyja-ah-backend.herokuapp.com/api/modules', {
      headers: { Authorization: `${localStorage.getItem('token')}` },
    });

    const { data } = modulesRequest.data;
    return dispatch({
      type: GET_ALL_MODULES,
      payload: data,
    });
  } catch (error) {
    return dispatch({
      type: MODULES_ERROR,
      payload: error,
    });
  }
};

export const setModule = payload => ({
  type: GET_SINGLE_MODULE,
  payload,
});

export const getModule = moduleId => async (dispatch) => {
  try {
    dispatch({ type: INIT_MODULE_REQUEST });
    const request = await get(`${baseUrl}/modules/${moduleId}`);
    const { data } = request.data;
    dispatch(setModule(data));
  } catch (error) {
    dispatch({
      type: MODULES_ERROR,
      payload: error,
    });
  }
};

export { getModules };
