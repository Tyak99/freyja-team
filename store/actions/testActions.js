import axios from 'axios';
import { GET_TEST, GET_ERRORS, INIT_GET_TEST } from './types';

const token = localStorage.getItem('token');
export const getTestSuccess = data => ({
  type: GET_TEST,
  payload: data,
});

export const setError = error => ({
  type: GET_ERRORS,
  errors: error,
});

export const getTest = moduleId => async (dispatch) => {
  const config = {
    headers: {
      Authorization: token,
    },
  };
  try {
    dispatch({
      type: INIT_GET_TEST,
    });
    const test = await axios.get(
      `https://freyja-ah-backend.herokuapp.com/api/tests/${moduleId}`,
      config,
    );
    if (test) {
      const { data } = test.data;
      dispatch(getTestSuccess(data));
    }
  } catch (error) {
    dispatch({
      type: GET_ERRORS,
      payload: error,
    });
  }
};
