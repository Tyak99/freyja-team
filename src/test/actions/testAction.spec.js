import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import { getTest } from '../../../store/actions/testActions';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe('test action', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });

  afterEach(() => {
    nock.cleanAll();
  });
  it('should test questions', async (done) => {
    nock('https://freyja-ah-backend.herokuapp.com')
      .get('/api/tests/1')
      .reply(200);
    await store.dispatch(getTest()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
