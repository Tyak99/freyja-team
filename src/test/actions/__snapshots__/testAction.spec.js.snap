// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test action should test questions 1`] = `
Array [
  Object {
    "type": "INIT_GET_TEST",
  },
  Object {
    "payload": [Error: Nock: No match for request {
  "method": "GET",
  "url": "https://freyja-ah-backend.herokuapp.com/api/tests/undefined",
  "headers": {
    "accept": "application/json, text/plain, */*",
    "authorization": {},
    "user-agent": "axios/0.19.0"
  }
}],
    "type": "GET_ERRORS",
  },
]
`;
